### ISSUE TITLE
Error code: 0x80072eff while executing `winget upgrade --all` command

### ISSUE THREAD

**Issue by sanamhub** [microsoft/winget-cli#1800]
### Brief description of your issue

0x80072eff : unknown error while executing `winget upgrade --all` command on my windows terminal

### Steps to reproduce

1. Open windows terminal
2. Paste `winget upgrade --all`
3. Hit enter

### Expected behavior

Update available updates for the packages

### Actual behavior

An unexpected error occurred while executing the command:
0x80072eff : unknown error

#### Screenshot

![image](https://user-images.githubusercontent.com/31341013/146117141-54267518-0093-4fb8-bd4e-f0076a1c51d3.png)

### Environment

```shell
Windows Package Manager v1.1.13405
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22000.376
Package: Microsoft.DesktopAppInstaller v1.16.13405.0

Note: I am running script on windows terminal
```

**Comment by gldraphael** [comment]
Similar issue here.

I had an old version of [PowerShell](https://github.com/PowerShell/PowerShell) installed through GitHub releases. 
Every time a new update was released, it required me to manually download PowerShell from GitHub releases.

I decided to try upgrading with `winget upgrade Microsoft.PowerShell`. It did not work, something about the existing installation using a different installation technology. I went ahead and uninstalled the existing installation from window's "Add or Remove Programs" interface.

And then tried installing with `winget install Microsoft.PowerShell`. Now I see one of two things:

```
0x80072eff : unknown error
```

Or I see it installing powershell indefinitely.

**Comment by sanamhub** [comment]
@denelon please have a look at this. (bug, 1+ year)

**Comment by denelon** [comment]
Can you share the output of `winget --info` and run the command with "--verbose-logs" and share the resulting log?
The log location is available in the output of `winget --info`.

Optionally from https://github.com/microsoft/winget-cli/releases/tag/v1.4.3132-preview and newer versions, we have a new experimental feature that may be enabled to open the directory with the logs.

**Comment by fsdrw08** [comment]
same to me, when I try to install microsoft.powershell by `winget install`, same error code throw out: `0x80072eff : unknown error`
```
Windows Package Manager v1.3.2691
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22621.1105
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.18.2691.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale
```

**Comment by denelon** [comment]
We've had several WinGet releases since this bug was reported. We've added some retry logic for download failures, and in some cases, the issue has been caused by defaulting to Delivery Optimization as a downloader in some enterprise environments and is transient or related to proxies in others. I'd like to know if this has been encountered on more recent versions of WinGet and if the error message has been improved.
